packages/phone-number-privacy/signer/src/domain/domain.service.ts

```diff
@@ -1,49 +1,75 @@
+import { Endpoints } from '../server'
 import { IDomainService } from './domain.interface'
-import { DOMAINS_COLUMNS } from '../database/models/domain'
-import { DomainStatusResponse } from './response/domainStatus.response'
 
 export class DomainService implements IDomainService {
-  constructor() {}
-
   public async handleDisableDomain(
-    request: Request<core.ParamsDictionary, {}, DisableDomainRequestBody>,
+    request: Request<{}, {}, DomainRequestBody>,
     response: Response
   ): Promise<void> {
     const logger = response.locals.logger
-    logger.info('Disabling domain', { domain: request.params.domain })
+    logger.info('Disabling domain', { domain: request.body.domain })
 
-    const disabled = request.body.value ?? true
-    const result = await setDomainDisabled(request.params.domain, disabled, logger)
-
-    if (result) {
-      response.sendStatus(200)
-    } else {
-      response.status(500).send({ message: 'Cannot disable the domain' })
-    }
+    return setDomainDisabled(request.body.domain, logger)
+      .then((result) => {
+        if (result) {
+          response.sendStatus(200)
+        } else {
+          respondWithError(
+            Endpoints.DISABLE_DOMAIN,
+            response,
+            500,
+            ErrorMessage.DATABASE_UPDATE_FAILURE
+          )
+        }
+      })
+      .catch((e) => {
+        logger.error('Error while disabling domain', e)
+        respondWithError(
+          Endpoints.DISABLE_DOMAIN,
+          response,
+          500,
+          ErrorMessage.DATABASE_UPDATE_FAILURE
+        )
+      })
   }
 
   public async handleGetDomainStatus(
-    request: Request<core.ParamsDictionary, {}, {}>,
+    request: Request<{}, {}, DomainRequestBody>,
     response: Response
   ): Promise<void> {
     const logger = response.locals.logger
-    logger.info('Getting domain status', { domain: request.params.domain })
-    const result = await getDomain(request.params.domain, logger)
-
-    if (result) {
-      const resultResponse: DomainStatusResponse = {
-        domain: result[DOMAINS_COLUMNS.domain],
-        counter: result[DOMAINS_COLUMNS.counter],
-        disabled: result[DOMAINS_COLUMNS.disabled],
-        timer: result[DOMAINS_COLUMNS.timer],
-      }
-      response.status(200).send(resultResponse)
-    } else {
-      response.status(500).send({ message: 'Cannot get status' })
-    }
+    logger.info('Getting domain status', { domain: request.body.domain })
+    getDomain(request.body.domain, logger)
+      .then((result) => {
+        if (result) {
+          const resultResponse: DomainStatusResponse = {
+            domain: result[DOMAINS_COLUMNS.domain],
+            counter: result[DOMAINS_COLUMNS.counter],
+            disabled: result[DOMAINS_COLUMNS.disabled],
+            timer: result[DOMAINS_COLUMNS.timer],
+          }
+          response.status(200).send(resultResponse)
+        } else {
+          respondWithError(
```

Comment:
We should probably tell the client that the domain does not exist in the db rather than return a 500.. Feel free to add a new error type in the common package for this

Reply:
agreed